src/metabase/lib/convert.cljc

```diff
@@ -70,8 +70,14 @@
   :hierarchy lib.hierarchy/hierarchy)
 
 (defn- default-MBQL-clause->pMBQL [mbql-clause]
-  (let [[clause-type options & args] (lib.options/ensure-uuid mbql-clause)]
-    (into [clause-type options] (map ->pMBQL) args)))
+  (let [last-elem (peek mbql-clause)
+        last-elem-option? (map? last-elem)
+        [clause-type & args] (cond-> mbql-clause
+                               last-elem-option? pop)
+        options (if last-elem-option?
+                  last-elem
+                  {})]
+    (lib.options/ensure-uuid (into [clause-type] (map ->pMBQL) (cons options args)))))
```

Comment:
I'm pretty sure `options` should not go through `->pMBQL`. The `update-vals` in `:dispatch-type/map` could easily make vector options be processed as an mbql clause.

Reply:
> The `update-vals` in `:dispatch-type/map` could easily make vector options be processed as an mbql clause.

Do you know which kinds of maps that branch is expected to process? (TBH, I'm a bit uneasy with our practice of defining functions that can take anything and convert to something...)

Reply:
@metamben  From what I've seen, I think it's largely about legacy MBQL and idempotency, but I can't speak to the intent.

Reply:
@snoe I went with your proposal. After all, the original didn't convert option values either.